addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -459,55 +457,26 @@ export class HtmlField extends Component {
-                            '<head>' +
-                                '<meta charset="utf-8"/>' +
-                                '<meta http-equiv="X-UA-Compatible" content="IE=edge"/>\n' +
-                                '<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"/>\n' +
-                            '</head>\n' +
-                            '<body class="o_in_iframe o_readonly" style="overflow: hidden;">\n' +
-                                '<div id="iframe_target"></div>\n' +
-                            '</body>' +
-                            '</html>');
-                }
-                if (this.props.cssReadonlyAssetId) {
-                    for (const cssLib of this.cssReadonlyAsset.cssLibs) {
-                        const link = cwindow.document.createElement('link');
-                        link.setAttribute('type', 'text/css');
-                        link.setAttribute('rel', 'stylesheet');
-                        link.setAttribute('href', cssLib);
-                        cwindow.document.head.append(link);
-                    }
-                    for (const cssContent of this.cssReadonlyAsset.cssContents) {
-                        const style = cwindow.document.createElement('style');
-                        style.setAttribute('type', 'text/css');
-                        const textNode = cwindow.document.createTextNode(cssContent);
-                        style.append(textNode);
-                        cwindow.document.head.append(style);
-                    }
-                }
-
-                if (!this.sandboxedPreview) {
-                    const iframeTarget = cwindow.document.querySelector('#iframe_target');
-                    iframeTarget.innerHTML = value;
-
-                    const script = cwindow.document.createElement('script');
-                    script.setAttribute('type', 'text/javascript');
-                    const scriptTextNode = document.createTextNode(
-                        `if (window.top.${this._onUpdateIframeId}) {` +
-                            `window.top.${this._onUpdateIframeId}(${_avoidDoubleLoad})` +
-                        `}`
+                    const iframeContent = await this.http.post(
+                        `/web_editor/wysiwyg_iframe`,
+                        {
+                            readonly: 1,
+                            content: value,
+                            bundleNames: this.props.cssReadonlyAssetId,
+                            avoidDoubleLoad: _avoidDoubleLoad,
+                            onUpdateIframeId: this._onUpdateIframeId,
+                            csrf_token: odoo.csrf_token,
+                        },
+                        "text"
                     );
-                    script.append(scriptTextNode);
-                    cwindow.document.body.append(script);
+                    cwindow.document.documentElement.innerHTML = iframeContent;
```

Comment:
@odoo/rd-security : It doesn't work with innerText because html is not interpolated.